File: piwigo/Image/Parameters/Cells/EditImagePrivacyTableViewCell.m
//
//  EditImagePrivacyTableViewCell.m
//  piwigo
//
//  Created by Spencer Baker on 2/16/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//

#import "EditImagePrivacyTableViewCell.h"
#import "Model.h"

@interface EditImagePrivacyTableViewCell()

@property (weak, nonatomic) IBOutlet UILabel *label;
@property (weak, nonatomic) IBOutlet UILabel *list;

@end

@implementation EditImagePrivacyTableViewCell

- (void)awakeFromNib {

    // Initialization code
    [super awakeFromNib];
	
    self.label.font = [UIFont piwigoFontNormal];
    self.list.font = [UIFont piwigoFontNormal];
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated
{
    [super setSelected:selected animated:animated];
    // Configure the view for the selected state
	
}

-(void)setLeftLabelText:(NSString*)text
{
    self.label.text = text;
    self.label.textColor = [UIColor piwigoColorLeftLabel];
}

-(void)setPrivacyLevel:(kPiwigoPrivacy)privacy inColor:(UIColor *)color
{
	self.list.text = [[Model sharedInstance] getNameForPrivacyLevel:privacy];
    self.list.textColor = color;
}

@end

File: piwigo/Image/Parameters/Cells/EditImagePrivacyTableViewCell.swift
//
//  EditImagePrivacyTableViewCell.swift
//  piwigo
//
//  Created by Spencer Baker on 2/16/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//
//  Converted to Swift 5.3 by Eddy Lelièvre-Berna on 29/08/2021.
//

import UIKit
import piwigoKit

class EditImagePrivacyTableViewCell: UITableViewCell {
    
    @IBOutlet private weak var label: UILabel!
    @IBOutlet private weak var list: UILabel!

    override func setSelected(_ selected: Bool, animated: Bool) {
        // Configure the view for the selected state
        super.setSelected(selected, animated: animated)
    }

    func setLeftLabel(withText text: String?) {
        label.text = text
        label.textColor = .piwigoColorLeftLabel()
    }

    func setPrivacyLevel(with privacy: kPiwigoPrivacy, inColor color: UIColor?) {
        list.text = privacy.name
        list.textColor = color
    }
}
